### ISSUE TITLE
NOT working

### ISSUE THREAD

**Issue by skandar2025** [hyperfield/yt-channel-downloader#13]
absolutely not fetching, not showing videos, not downloading!

**Comment by hyperfield** [comment]
To troubleshoot this issue can you please provide the following information:

- Your operating system.
- Did the previous version work (if you've tried it)?

**Comment by hyperfield** [comment]
> absolutely not fetching, not showing videos, not downloading!

To troubleshoot this issue can you please provide the following information:

  - Your operating system.
  - Did the previous version work (if you've tried it)?

**Comment by skandar2025** [comment]
Thx for hyperfield, for your prompt answer!

* Windows 11
* Nothing absolutely happens, fetching not working, download not working when i click on the botton.
* I used both, source and installer (generally i love installer more easier)
* do you believe? i used all versions and all of them are not working.

Kindly do fix this issue asap, and inform me please

**Comment by hyperfield** [comment]
Did you install `ffmpeg` as described in the README?

**Comment by skandar2025** [comment]
Yes, exactly!

**Comment by hyperfield** [comment]
Can you please let me know if your country blocks or slows down YouTube?

**Comment by skandar2025** [comment]
Yes, in a country that Youtube is slowed down, but i open youtube using VPN (Planet VPN) , and it's working fast. but i used vpn and without vpn your program but still the same! I told you it seems that the issue is in your program actually!

**Comment by hyperfield** [comment]
The VPN that you're using, is it a browser VPN or does it change your entire system IP address? You can find out by running the following command in PowerShell `curl https://ipinfo.io/ip` and observing the given IP address.

**Comment by skandar2025** [comment]
Brother, of course not a vpn browser! and it is showing other location country, and in that country youtube is working perfectly. and yes VPN changing my entire system!

**Comment by hyperfield** [comment]
For debugging reasons, just to make sure, if you run the following Python code, do you get your VPN's IP address?

```
import urllib.request

with urllib.request.urlopen('https://ipinfo.io/ip') as response:
    public_ip = response.read().decode('utf-8').strip()
    print(f"Public IP Address: {public_ip}")
```

**Comment by skandar2025** [comment]
Here is the result
![Capture d’écran 2024-09-30 221719](https://github.com/user-attachments/assets/3c12a008-f5ac-48c6-9a93-58a9fa71e917)

**Comment by hyperfield** [comment]
[ip.zip](https://github.com/user-attachments/files/17196817/ip.zip)
Please run the included .py file with `python ip.py`.

**Comment by skandar2025** [comment]
Here you go:

![Uploading Capture d’écran 2024-10-01 011056.png…]()

**Comment by skandar2025** [comment]
So what's going on bro?

**Comment by digifjord** [comment]
I've conducted some tests, and the app works fine when YouTube is neither blocked nor slowed down, but it hangs otherwise (which will need to be handled gracefully with programmatic exceptions). I can't yet determine the exact issue in your case, but the next step would be to try using the app with a SOCKS proxy server.
